### PR TITLE
[Enhancement] Support loading GT for evaluation from multi-file backend

### PR DESCRIPTION
**Pull request by uni19**
## Motivation
support load gt for evaluation from multi file backend

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by Junjun2016 — COMMENTED** [review]
LGTM

File: mmseg/datasets/custom.py [hunk 1]
```diff
@@ -232,6 +239,14 @@ def format_results(self, results, imgfile_prefix, indices=None, **kwargs):
         """Place holder to format result to dataset specific output."""
         raise NotImplementedError
 
+    def get_one_gt_seg_map(self, index):
```
Comment (xvjiarui) [inline mmseg/datasets/custom.py:242]:
```suggestion
    def get_gt_seg_map_by_idx(self, index):
```

**Review by Junjun2016 — APPROVED** [review]
LGTM

**Review by xvjiarui — APPROVED** [review]

**Comment by xvjiarui** [comment]
@uni19 Thanks for your nice contribution!

**Pull request merged**